### PR TITLE
use ? to simplify `TransitiveRelation.maybe_map`

### PR DESCRIPTION
**Pull request by llogiq**
I think this looks much clearer than the original.

### CODE REVIEW

File: src/librustc_data_structures/transitive_relation.rs [hunk 1]
```diff
@@ -59,6 +59,7 @@ impl<T: Clone + Debug + Eq + Hash> TransitiveRelation<T> {
             edges: vec![],
             closure: Lock::new(None),
         }
+
```
Comment (llogiq) [inline src/librustc_data_structures/transitive_relation.rs]: Oops, I added this by mistake.

**Comment by joshtriplett** [comment]
@llogiq Please don't backmerge master into your feature branch. If you need to update your feature branch, please rebase it instead.

**Comment by joshtriplett** [comment]
For some reason, when I try to review lines of your commit, I get this:
> Start commit oid is not part of the pull request

In any case: please drop the change to `add_index`, which seems unrelated to the change described in the commit message.

Apart from that, this looks great.

**Comment by llogiq** [comment]
This is what happens when one uses the UI (I was on mobile earlier). The history should now be fixed.

**Comment by llogiq** [comment]
@joshtriplett r?

**Pull request merged**